docs/rag.md

```diff
@@ -0,0 +1,36 @@
+
+The key advantage of RAG is its ability to leverage external knowledge to generate answers, which allows it to provide more detailed and accurate responses compared to models that rely solely on pre-trained knowledge. It's particularly effective in question-answering tasks and can be used in a wide range of applications, including chatbots, customer service, and AI research.
+
+In the context of Azure AI, while Azure OpenAI API does provide an out-of-the-box RAG solution, you can also leverage the suite of AI and machine learning tools to implement such a model. Azure's Machine Learning service provides the infrastructure and tools necessary for training, testing, and deploying machine learning models. You can also use Azure's Document Intelligence API for processing and analyzing text data, and Azure Web App Service for creating conversational AI applications.
+
+## Azure AI Search
+
+Azure AI Search and OpenAI can be paired to create a powerful search and AI solution. OpenAI provides advanced AI models, like GPT-4, which can generate human-like text. Azure AI Search provides sophisticated search capabilities. Here's how they could work together:
+Data Ingestion and Indexing: Azure AI Search ingests data from various sources like Azure SQL Database, Azure Cosmos DB, Azure Blob Storage, and more. It then creates an index based on the data fields defined by the user, which makes the search operations faster and more efficient.
+Enrichment with OpenAI: The ingested data can be sent to OpenAI's GPT-4 model for enrichment. For example, GPT-4 could generate summaries of long documents, answer questions about the content, or extract key points. This enriched data can then be indexed by Azure AI Search, enhancing the search capabilities further.
+Querying with OpenAI: When a query is made, it could be processed by OpenAI's GPT-4 to understand the context better. For example, GPT-4 could convert a natural language query into a more structured form that Azure AI Search can understand.
+Search and Retrieval: Azure AI Search retrieves the most relevant results based on the processed query. The results are ranked according to relevance, providing a rich search experience.
+Post-processing with OpenAI: The search results can be further processed by OpenAI's GPT-4 for better presentation. For example, GPT-4 could generate a natural language summary of the search results.
+Monitoring and Maintenance: The performance and usage of the search service can be tracked using Azure Monitor and Azure Log Analytics. The search indices, data sources, and other resources can be managed using the Azure portal, REST APIs, or .NET SDK.
+
+## RAG pattern on Azure
```

Comment:
Would it make sense to add numbers (1, 2, ... N) to the picture to match the points below?

Reply:
Working on that. Will be added in the next update.